Add a new option to WebFrontAuthOptions : SlidingExpirationRefreshLimit:

Purgator opened the pull request:
If set to true, it will behave kind of like dotnet cookie.
It means that the expiration time will be refreshed only when it processes a request which is more than halfway through the expiration window.
Since ExpirationTimeSpan and SlidingExpirationTime exists, this code will be ignored and throw a warning whenever those 2 values are not equals.

This commit has been issued to solve antiforgery issues.

Purgator committed:
Add a new option to WebFrontAuthOptions : SlidingExpirationRefreshLimit:

If set to true, it will behave kind of like dotnet cookie.
It means that the expiration time will be refreshed only when it processes a request which is more than halfway through the expiration window.
Since ExpirationTimeSpan and SlidingExpirationTime exists, this code will be ignored and throw a warning whenever those 2 values are not equals.

This commit has been issued to solve antiforgery issues.

Purgator committed:
Cleanup.

Purgator committed:
ActivityMonitor should be the first parameter.

Purgator committed:
Fix Couldn't allocate enough memory by forcing appveyor to use Node x64.

Purgator committed:
Fix typo.

bcrosnier commented:
Might want to add some UTs. SlidingExpiration is one of the things tested (in `WebFrontHandlerTests`), I'd love to keep it that way.

I'll add that the referenced .NET Core property is [CookieAuthenticationOptions.SlidingExpiration](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.cookies.cookieauthenticationoptions.slidingexpiration?view=aspnetcore-6.0).

Purgator committed:
Unit test new property SlidingExpirationRefreshLimit:

A refresh should extend life time only on its second half.
Also, a different lifetime between ExpireTimeSpan and SlidingExpirationTime cancel this behavior.

Purgator committed:
Add reference to dotnet Cookie implementation that inspired the last change on SlidingExpirationRefreshLimit.

Purgator commented:
Since we discussed about the way to handle AntiForgery Token in the case of an OIDC flown this PR is getting closed.

The main idea is to keep a **Single point of truth**. Since the frontend call the refresh, we do not want the backend to decide again if it should actually refresh.
We notice that we need to handle the AntiForgery specifically for the OIDC scenario.
Hence, we want to let the frontend handle it.

Here is the solution :

```
    if( !window.location.hash.startsWith( '#/auth-consent?' ) )
      await authService.refresh( false, true, true );
```

in `src/app/app.initializer.ts`

Considering '#auth-consent' is the path for the consent form.
